src/Smart.Design.Razor/TagHelpers/Elements/Checkbox/CheckboxTagHelper.cs

```diff
@@ -64,7 +65,11 @@ public CheckboxTagHelper(ICheckboxHtmlGenerator checkboxHtmlGenerator)
     /// <inheritdoc />
     public override void Process(TagHelperContext context, TagHelperOutput output)
     {
-        var inputCheckbox = _checkboxHtmlGenerator.GenerateCheckbox(Id, Name, Value, Label, Disabled, Checked, For);
+        var attributeObjects = context.AllAttributes.ToList();
+        var properties = GetType().GetProperties().Select(p => p.Name.ToLower());
+        attributeObjects.RemoveAll(a => properties.Contains(a.Name));
```

Comment:
That will be probably be used a lot in the future (unfortunately because of properties override...). Could we create an extension method to list undefined attribute? Something like: 
```
public static class AttributeHelperExtensions
{
    /// <summary>
    /// List All extra attributes which are not explictely defined properties on the custom tag helper implementation
    /// </summary>
    /// <param name="obj">Instance of the tag helper (this)</param>
    /// <param name="context">Context of the tag helper</param>
    /// <typeparam name="T">Type of the tag helper</typeparam>
    /// <returns></returns>
    public static List<TagHelperAttribute> ListUndefinedTagHelperAttributes<T>(this T obj, TagHelperContext context) where T: BaseTagHelper
    {
        var type = typeof(T);
        var attributeObjects = context.AllAttributes.ToList();
        var properties = type.GetProperties().Select(p => p.Name.ToLower());
        attributeObjects.RemoveAll(a => properties.Contains(a.Name));
        return attributeObjects;
    }
}
```

Then those lines could be replaced by:
```suggestion
        var attributeObjects = this.ListUndefinedTagHelperAttributes(context);
```

(You might have a cleaner/different way, as long as that is reusable I guess)